artichoke-backend/src/extn/core/encoding/mod.rs

```diff
@@ -0,0 +1,33 @@
+//! Encoding represents a character encoding usable in Ruby
+//!
+//! This module implements the [`Encoding`] class from Ruby Core.
+//!
+//! You can use this class in your application by accessing it directly. As a
+//! Core class, it is globally available:
+//!
+//! ```ruby
+//! Encoding.list
+//! ```
+//!
+//! [`Encoding`]: https://ruby-doc.org/3.1.2/Encoding.html
+
+pub(in crate::extn) mod mruby;
+
+struct Encoding {}
+
+#[cfg(test)]
+mod tests {
+    use crate::test::prelude::*;
+
+    const SUBJECT: &str = "Encoding";
+    const FUNCTIONAL_TEST: &[u8] = include_bytes!("encoding_test.rb");
+
+    #[test]
+    fn functional() {
```

Comment:
I do love this pattern and love when new functional test suites get added. thanks for adding this.